node/src/components/consensus/highway_core/state.rs

```diff
@@ -517,21 +517,26 @@ impl<C: Context> State<C> {
 
         // Create an Evidence instance for each conflict we found.
         conflicting_endorsements
-            .map(|(vidx, uhash1, sig1, uhash2, sig2)| {
+            .filter_map(|(vidx, uhash1, sig1, uhash2, sig2)| {
                 let unit1 = self.unit(uhash1);
+                let unit2 = self.unit(uhash2);
+                let ee_limit = self.params().endorsement_evidence_limit();
+                if unit1.seq_number.saturating_add(ee_limit) < unit2.seq_number {
+                    return None;
+                }
```

Comment:
So, this could well be wrong, but I'm trying to understand how the `ee_limit` works.
We have two conflicting endorsements `unit1` and `unit2`.

If `unit1.seq_number` is _way below_ `unit2.seq_number`, then we don't bother. 

Is this because we know there will be another pair of conflicting endorsements, so we can limit the set we keep?

Reply:
No, it's just because we want to limit the size of our messages somehow, so we actually let you get away with that kind of fault.
We also know that sequence numbers are bounded by a limit that grows linearly in time, so if we make our endorsement evidence limit large enough we know you can't make the kind of fault that wouldn't get slashed until the era is likely to be over.

Reply:
Thank you for the clarification.